src/color.rs

```diff
@@ -141,7 +141,7 @@ impl Color {
     ///
     /// FIXME(#2) Deprecated CSS2 System Colors are not supported yet.
     pub fn parse<'i, 't>(input: &mut Parser<'i, 't>) -> Result<Color, BasicParseError<'i>> {
-        let token = input.next()?;
+        let token = input.next()?.clone();
```

Comment:
is this clone needed? couldn't you just use `match *token`? I guess maybe you can't parse the nested block below, is that it?

And, you need the clone below for the error?